src/mono/System.Private.CoreLib/src/System/TypedReference.Mono.cs

```diff
@@ -16,6 +16,21 @@ public ref partial struct TypedReference
         #pragma warning restore CA1823
         #endregion
 
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private static TypedReference MakeTypedReferenceInternal(object target, IntPtr[] fields, RuntimeType lastFieldType)
+        {
+            TypedReference result = default;
+            unsafe
+            {
+                InternalMakeTypedReference(&result, target, fields, lastFieldType);
+            }
+            return result;
```

Comment:
I converted this to totally managed:

```csharp
return new TypedReference(ref Unsafe.Add(ref target.GetRawData(), offset), targetType.m_handle);
```

Where `offset` is produced by `offset += RuntimeFieldHandle.GetInstanceFieldOffset(field);`.

Reply:
This is incorrect. I misses EnC scenarios.